pages/understanding-json-schema/reference/combining.md

```diff
@@ -135,6 +135,8 @@ Multiple of *both* 5 and 3 is rejected.
 15
 ```
 
+Careful consideration should be taken when using `oneOf` entries as the nature of it requires verification of all sub-schemas so doing anything recursive can have a severe performance impact.
```

Comment:
As I mhad mentioned in the issue, the concern is not recursive schemas.  Yes, that can increase iterations, but the primary thing here is that `oneOf` _requires_ evaluating all of the schemas.  The preference is `anyOf` when possible.

I think that separately, recursive schemas should have a warning.

---

Also, I'd be interested in what recursive reference inside a `oneOf` is.  Typically, this will result in the kind of recursion that's disallowed by the spec, e.g. `{ "$ref": "#" }` as this produces infinite loops.

Reply:
> I think that separately, recursive schemas should have a warning.

I agree. A general warning about recursive schemas is warranted, and a separate one for the use of `oneOf`.

> Also, I'd be interested in what recursive reference inside a oneOf led you to file the issue initially

It's been a while since I opened that initial issue (https://github.com/justinrainbow/json-schema/issues/640) but the TLDR is that it's easy to make a schema that takes a long time to process, and it's not immediately obvious where the issue is.

Reply:
Oh, okay.  Yeah that schema is a lot like the notorious FHIR schema, which has been jokingly used as a benchmark among us library maintainers.  A lot of implementations just choke on it (including my previous implementation).

It has around 50 definitions and a `oneOf` that references most (if not all) of them, then a lot of the definitions have nested references back to the root.  It's horrible.  People shouldn't do that.

If their data looks like that, I'd say they have a larger problem with data design, and JSON Schema isn't going to help the situation.  In these cases, when JSON Schema exacerbates a problem, it's better to find the root cause and fix that rather than throwing another tool at it.